### PR TITLE
Break test.js into small pieces

### PR DESCRIPTION
**Pull request by zhongyi-zhang**
This PR doesn't change any logic in the original test framework. It almost does only "moving".
With the change,
* It is easier to maintain.
* https://github.com/openservicebrokerapi/osb-checker/issues/21 can take advantages of this change, too. A big problem in that issue is how to reuse existing test cases. A new parameter `apiVersion` is added to all the common test functions which are moved from test.js.

### CODE REVIEW

File: common/testAsyncParameter.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+function testAsyncParameter (handler, verb, body, apiVersion) {
+  if (!apiVersion) {
+    apiVersion = '2.13'
+  }
+  if (apiVersion === '2.13') {
+    config = require('../2.13/tests/test/configs/config_mock.json')
+    preparedRequest = require('../2.13/tests/test/preparedRequest')
+  } else {
+    // supprt 2.14 here
+    throw Error('testProvision doesn\'t support this api version')
```
[the remark this answers is not in the document]
Comment (zhongyi-zhang) [reply common/testAsyncParameter.js]: Yeah, thanks for catching it. Other test*.js have the same issue.

**Comment by zhongyi-zhang** [comment]
The entry point is not changed.

File: common/testAPIVersionHeader.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/* eslint-env node, mocha */
+var config
+var preparedRequest
+
+function testAPIVersionHeader (handler, verb, apiVersion) {
+  if (!apiVersion) {
+    apiVersion = '2.13'
+  }
+  if (apiVersion === '2.13') {
```
Comment (leonwanghui) [inline common/testAPIVersionHeader.js]: Sorry, but I don't get it how this framework can be extended easily, maybe we could consider `gemver`?

Comment (zhongyi-zhang) [reply common/testAPIVersionHeader.js]: Considering OSB API is a back compatible API, we can write something like
```
if (semver.ge(apiVersion, '2.13') {
  run 2.13 cases here
}
if (semver.ge(apiVersion, '2.14') {
  run 2.14 new cases here
}
```
And `test.js` under the folder `2.14` also import the functions in under `common`.
Whatever, it should be another item in another PR. This PR wouldn't do that.

**Pull request merged**